tests/test_metrics.py

```diff
@@ -167,6 +167,53 @@ def test_xpath_all_with_header(self):
         for b in ["foo", "bar", "baz"]:
             self.assertSequenceEqual(ret[b]["metric_hcsv"], [[b]])
 
+    def test_formatted_output(self):
+        import csv
+
+        values = [
+            ["value_mse", "deviation_mse", "data_set"],
+            ["0.421601", "0.173461", "train"],
+            ["0.67528", "0.289545", "testing"],
+            ["0.671502", "0.297848", "validation"],
+        ]
+
+        with open("metrics.csv", "w") as fobj:
+            csv.writer(fobj).writerows(values)
+
+        with open("metrics.tsv", "w") as fobj:
+            csv.writer(fobj, delimiter="\t").writerows(values)
+
+        self.dvc.run(
+            fname="testing_metrics_output.dvc",
+            metrics_no_cache=["metrics.csv", "metrics.tsv"],
+        )
+
+        self.dvc.metrics.modify("metrics.csv", typ="csv")
+        self.dvc.metrics.modify("metrics.tsv", typ="tsv")
+
+        with MockLoggerHandlers(logger.logger):
+            reset_logger_standard_output()
+
+            ret = main(["metrics", "show"])
+
+            expected_output = (
```

Comment:
love the test! please, let's add a simple test for an edge case - no delimiters or no newlines

Reply:
I'm missing this one, will work on it tomorrow :)